Ghidra/Processors/6800/data/languages/6800.slaspec

```diff
@@ -0,0 +1,459 @@
+        setNZFlags(reg);
+}
+
+macro compare(reg, op) {
+        $(V) = sborrow(reg, op);
+        local tmp = reg - op;
+        setNZFlags(tmp);
+        $(C) = (tmp < op);
+}
+
+macro compare16(reg, op) {		# used for 16 bit CPX; C not affected
+        $(V) = sborrow(reg, op);
+        local tmp = reg - op;
+        setNZFlags(tmp);
+}
+
+macro logicalAnd(reg, op) {
+        reg = reg & op;
+        setNZFlags(reg);
+        $(V) = 0;
+}
+
+macro logicalOr(reg, op) {
+        reg = reg | op;
+        setNZFlags(reg);
+        $(V) = 0;
+}
+
+macro logicalExclusiveOr(reg, op) {
+        reg = reg ^ op;
+        setNZFlags(reg);
+        $(V) = 0;
+}
+
+macro bitTest(reg, op) {
+        local tmp = reg & op;
+        setNZFlags(tmp);
+        $(V) = 0;
+}
+
+macro loadRegister(reg, op) {
+        reg = op;
+        setNZFlags(reg);
+        $(V) = 0;
+}
+
+macro storeRegister(reg, op) {
+        op = reg;
+        setNZFlags(reg);
+        $(V) = 0;
+}
+
+macro Push1(reg, op) {		# 1 byte operand
+        *:1 reg = op;
+        reg = reg - 1;
+}
+
+macro Push2(reg, op) {		# 2 byte operand
+        reg = reg - 1;
+        *:2 reg = op;
+        reg = reg - 1;
+}
+
+macro Pull1(reg, op) {
+        op = *:1 reg;
+        reg = reg + 1;
+}
+
+macro Pull2(reg, op) {
+        reg = reg + 1;
+        op = *:2 reg;
+        reg = reg + 1;
+}
+macro PushEntireState() {
+		local addr:2 = inst_next + 1; 
+		Push2(SP, addr);
+		Push2(SP, X);	
+        Push1(SP, A); 
+        Push1(SP, B); 
+        Push1(SP, CC);
+}
+macro PullEntireState() {
+		local addr:2; 
+		Pull1(SP, CC); 
+		Pull1(SP, B);
+        Pull1(SP, A); 
+        Pull2(SP, X); 
+        Pull2(SP, addr); 
+        return [addr];
+}
+
+################################################################
+# Addressing tables
+################################################################
+
+#	Branches and jumps addresses
+OPREL: addr     is simm8  [ addr = inst_start + simm8 + 2;  ]   { export *:2 addr; } 			# relative
+OPJ: imm8",X" 	is (ophi=0x6 | ophi=0xA); imm8 { addr:2 = X + imm8; export *:2 addr; } 			# indexed
+OPJ: imm16 		is (ophi=0x7 | ophi=0xB); imm16 { export *:2 imm16; } 							# extended
+
+#	OPAB  = operand for operations on accumulators A or B
+OPAB: "#"^imm8  is (ophi=0x8 | ophi=0xC); imm8 { export imm8; } 								# immediate
+OPAB: imm8    	is (ophi=0x9 | ophi=0xD); imm8 { export *:1 imm8; } 							# direct
+OPAB: imm8",X"  is (ophi=0xA | ophi=0xE); imm8 { local addr:2 = X + imm8; export *:1 addr; } 	# indexed
+OPAB: imm16  	is (ophi=0xB | ophi=0xF); imm16 { export *:1 imm16; } 							# extended
+
+#   OPM   = for operations working on memory locations
+OPM:  imm8",X"  is ophi=0x6; imm8 { local addr:2 = X + imm8; export *:1 addr; } 				# indexed
+OPM:  imm16  	is ophi=0x7; imm16 { export *:1 imm16; } 										# extended
+
+#	OP16  = for operations on 16 bit registers SP X
+OP16: "#"^imm16 is (ophi=0x8 | ophi=0xC); imm16 { export imm16; } 								# immediate
```

Comment:
just being bit by issues that look like this I would  `export imm16:2`

Reply:
The compiler wouldn't take this.

Reply:
Ah, oops, guess it would just be  a `local tmp:2 = imm16; export tmp;` and `local tmp:1 = imm8; export tmp;` for the other

Reply:
`export *[const]:2 imm16` would be best.